spec/cases/graph_api_batch_spec.rb

```diff
@@ -406,6 +416,13 @@
               Koala::Facebook::API.new("foo").batch {|batch_api| batch_api.get_object('me') }
             }.to raise_exception(Koala::Facebook::BadFacebookResponse)
           end
+          
+          it "raises a BadFacebookResponse if the body is non-empty, non-array" do
+            allow(Koala).to receive(:make_request).and_return(Koala::HTTPService::Response.new(200, "200", {}))
+            expect {
+              Koala::Facebook::API.new("foo").batch(http_component: :response) {|batch_api| batch_api.get_object('me') }
```

Comment:
Could you add a test where http_component is status, and another one with headers

Reply:
Sure. We have added tests for http_component `:status` and `:headers` as well.